File: client/src/game/conf/SlotsConfig.ts
class SlotsConfig {
    private static _dic:Dictionary<SlotsConfig> = new Dictionary<SlotsConfig>();
    private static _arr:Array<SlotsConfig> = new Array<SlotsConfig>();

	 public Id :number ;
     public Name :string ;
     public BeforeSlot : Array<number>  =new Array<number>();
    
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<SlotsConfig>{
        return  SlotsConfig._dic;
    }
    public static get Arr():Array<SlotsConfig>{
        return SlotsConfig._arr;
    }
    public static parse(json:any[]){
        if(json && json.length > 0){
			for(var i = 0; i < json.length; i++){
				let config:SlotsConfig = <SlotsConfig>json[i];
				SlotsConfig._arr.push(config);
				SlotsConfig._dic.add(config.Id,config);
			}
        }else{
            console.error("SlotsConfig loader fail!");
        }
    }
}

File: tools/conf/output/go/SlotsConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type SlotsConfigElem struct {
	Id int
	Name string
	BeforeSlot []int
	Des string
	
}

type SlotsConfig0 struct {
	Dic map[int]*SlotsConfigElem
	Arr []*SlotsConfigElem
}

var (
	SlotsConfig SlotsConfig0
)

func loadSlotsConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	SlotsConfig.Dic = make(map[int]*SlotsConfigElem)
	SlotsConfig.Arr = make([]*SlotsConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &SlotsConfig.Arr)
	for _,conf := range SlotsConfig.Arr{
		SlotsConfig.Dic[conf.Id] = conf
	}
	
}
